File: app/src/main/java/com/example/fluffstroller/models/PaymentData.java
package com.example.fluffstroller.models;

public class PaymentData {
    private String currency;
    private String ownerId;
    private String strollerId;
    private String walkId;
    private Double amount;

    public PaymentData() {
    }

    public PaymentData(String currency, String ownerId, String strollerId, String walkId, Double amount) {
        this.currency = currency;
        this.ownerId = ownerId;
        this.strollerId = strollerId;
        this.walkId = walkId;
        this.amount = amount;
    }
}

File: backend/location_tracker/src/paymentData.ts
export interface PaymentData {
    currency: string;
    amount: number;
    metadata: PaymentDataMetadata;
}

type PaymentDataMetadata = WalkPayMetadata | StrollerPayMetadata;

interface WalkPayMetadata {
    ownerId: string;
    strollerId: string;
    walkId: string;
}

interface StrollerPayMetadata {
    userId: string;
}
